nucypher/policy/conditions/address.py

```diff
@@ -0,0 +1,147 @@
+import json
+from typing import Dict, List, Optional, Tuple, Union
+
+from eth_utils import to_checksum_address
+from marshmallow import fields, validate
+
+from nucypher.policy.conditions.base import AccessControlCondition
+from nucypher.policy.conditions.context import (
+    USER_ADDRESS_CONTEXT,
+    resolve_any_context_variables,
+)
+from nucypher.policy.conditions.exceptions import (
+    InvalidCondition,
+    InvalidConditionContext,
+)
+from nucypher.policy.conditions.lingo import (
+    ConditionType,
+)
+
+
+class AddressAllowlistCondition(AccessControlCondition):
+    """
+    A condition that checks if a user's wallet address is in a list of allowed addresses.
+    The user must provide a signed message to prove ownership of the wallet.
+    """
+
+    CONDITION_TYPE = ConditionType.ADDRESS_ALLOWLIST.value
+
+    class Schema(AccessControlCondition.Schema):
+        condition_type = fields.Constant(ConditionType.ADDRESS_ALLOWLIST.value)
+        addresses = fields.List(
+            fields.String(required=True),
+            required=True,
+            validate=validate.Length(min=1, max=25),
+        )
+
+    def __init__(
+        self,
+        addresses: List[str],
+        name: Optional[str] = None,
+    ):
+        """
+        Initialize a AddressAllowlistCondition.
+
+        Args:
+            addresses: List of checksummed wallet addresses that are allowed to decrypt (must be
+             properly checksummed Ethereum addresses or an exception will be raised)
```

Comment:
WDYT about this wording? Just trying to not use the word wallet and to not give redundant explanations.
```suggestion
            addresses: List of checksummed Ethereum addresses that are allowed to decrypt
```